### PR TITLE
Add MSAA sample

### PR DESCRIPTION
**Pull request by JoseEmilio-ARM**
Adds a new performance sample that shows the benefits of multisample anti-aliasing (MSAA) and how to resolve the multisampled attachments with minimum impact on performance.

The UI controls allow the user to choose between different levels of MSAA and to select whether or not to resolve the color and depth attachments within the render pass.

This sample also shows how to use `VK_KHR_depth_stencil_resolve` to resolve depth and sample from it in a simple postprocessing use-case.

More details can be found in the sample tutorial: https://github.com/KhronosGroup/Vulkan-Samples/blob/c1b5e0f0e458c6b9a29611b668f64c085715bb0b/samples/performance/msaa/msaa_tutorial.md

Tested on Windows 10 (GTX 1080) and a Galaxy S10 (Mali G76).

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes build and run on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [x] I have tested the sample on at least one compliant Vulkan implementation
- [x] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [x] Any dependent assets have been merged and published in downstream modules

### CODE REVIEW

**Comment by SaschaWillems** [comment]
Sample and tutorial are looking great :)

It works fine for me on Desktop (RTX 2060), but on a low-spec Android device with an Adreno 505 it crashes. Looking through the logs it seems that the sample runs into a driver deadlock and then throws a timeout when trying to render the first frame. I'm not getting any validation layer message, so this may be caused by a device limitation.

Not sure if we should support such low specced devices though.

**Review by mattruschnv — DISMISSED** [review]
Runs fine on my RTX 2070. The debug camera starts in a little bit of a weird space, but that's very minor.

**Review by gary-sweet — DISMISSED** [review]
Works for Broadcom

**Comment by JoseEmilio-ARM** [comment]
> Sample and tutorial are looking great :)
> 
> It works fine for me on Desktop (RTX 2060), but on a low-spec Android device with an Adreno 505 it crashes. Looking through the logs it seems that the sample runs into a driver deadlock and then throws a timeout when trying to render the first frame. I'm not getting any validation layer message, so this may be caused by a device limitation.
> 
> Not sure if we should support such low specced devices though.

Thanks Sascha! I reproduced the crash and fixed it in #71 . I'll rebase this PR as soon as we get all required approvals to minimise the number of reviews.

File: samples/performance/msaa/msaa_tutorial.md
Comment (JoseEmilio-ARM) [inline samples/performance/msaa/msaa_tutorial.md]: Here we could show the Vulkan call rather than the framework

**Review by MichaelParkinWhite — DISMISSED** [review]
Looks good to me!

File: samples/performance/msaa/msaa_tutorial.md
Comment (JoseEmilio-ARM) [reply samples/performance/msaa/msaa_tutorial.md]: Done

**Review by mattruschnv — APPROVED** [review]
Works great

**Review by aurolou — APPROVED** [review]

**Review by gary-sweet — APPROVED** [review]

**Pull request merged**
